src/progress.rs

```diff
@@ -226,6 +252,85 @@ impl ProgressSet {
             self.progress.len()
         );
     }
+
+    /// Returns the minimal committed index for the cluster.
+    ///
+    /// Eg. If the matched indexes are [2,2,2,4,5], it will return 2.
+    pub fn minimum_committed_index(&self) -> u64 {
```

Comment:
I think it returns the latest commit index(in leader's view).

Reply:
I used the name from the old code here to avoid making too much changes, but I agree the name is misleading. How about `quorum_committed_index()`, `index_commited_by_quorum()`, or `known_commited_index()`?

I'd note this is different than the leader's last committed index. It needs to be the leader's view of what a quorum has committed to. The leader may have sent a new commit index to peers and only some may have informed it that they've committed to that index.

For example, if the leader (1) sent the commit index to other nodes (2, 3) which both got the message and then both 2/3 go down before they can respond, we still need this to return a 1.

At least according to my understanding of the function's requirements.

Reply:
`matched` is the index of the highest known matched entry. It does not means the committed index. A entry is committed once quorum has persisted it. How about `maximal_committed_index`?

Following links maybe helpful.
https://github.com/pingcap/raft-rs/blob/3799cfa482e288a0fc4a3a6c365c6681afd85b3a/src/raft_log.rs#L417-L425
https://github.com/etcd-io/etcd/blob/master/raft/design.md#progress

Reply:
I think it's a good name, I''ll reflect that